h2o-core/src/main/java/water/parser/PreviewParseWriter.java

```diff
@@ -18,7 +20,7 @@
   protected int _invalidLines;
   private   String []   _colNames;
   protected String [][] _data = new String[MAX_PREVIEW_LINES][];
-  private IcedHashMap<String,String>[] _domains;  //used in leiu of a HashSet
+  private Collection<String>[] _domains;
```

Comment:
why not just String[][]?

Reply:
`PreviewParseWriter` needs to use internally a hash map or hash set for effective insertion in the [`addStrCol`](https://github.com/h2oai/h2o-3/pull/5065/files#diff-1f9b998f1c157b03f6fb652a49fa1460a79ec7a30e569563579f0d7326d027b4L107) method and calculation of distinct column values (a domain of a categorical column).

In this PR, I want to extend `PreviewParseWriter` for another use case. In SW use case, [the domain is already known](https://github.com/h2oai/sparkling-water/blob/68a8a90d4e3c9242fb5109d58be3585da2da89ea/extensions/src/main/scala/ai/h2o/sparkling/extensions/rest/api/ImportFrameHandler.scala#L99), and doesn't have to be calculated via the `addStrCol` method. In SW, I just want to pass the domain to `PreviewParseWriter`  as a string array and avoid allocation of a big hash map when domain cardinality is high.

I've chosen `Collection` as a type for domains since it's a common ancestor for a hash set and a list which will be backed by the original string array representing the domain.